src/main.js

```diff
@@ -110,28 +114,44 @@ class Entry {
  * @returns {Array} An array containing entry objects.
  */
 const parseVarFromType = envSpecAsArray => {
-  return (envSpecAsArrayParsed = envSpecAsArray.map(element => {
-    //in case of typed variable or given restricted choices
-    if (element.includes(":")) {
-      element = element.split(":");
-      //if there is a default value indicated by an existing "="
-      if (element[1].includes("=")) {
-        element[1] = element[1].split("=");
+  return (envSpecAsArrayParsed = envSpecAsArray
+    .map(element => {
+      comment = "";
+      //in case of existing comment ignore what's after the "#" symbol
+      if (element.includes("#")) {
+        separatedComFromElem = element.split("#");
+        element = separatedComFromElem[0];
+        comment = separatedComFromElem[1];
+      }
+      //in case of typed variable or given restricted choices
+      if (element.includes(":")) {
+        element = element.split(":");
+        //if there is a default value indicated by an existing "="
+        if (element[1].includes("=")) {
+          element[1] = element[1].split("=");
+          return new Entry(
+            element[0].trim(),
+            element[1][0].trim(),
+            null,
+            element[1][1].trim(),
+            comment
+          );
+        }
+        //else if there is just a type
         return new Entry(
           element[0].trim(),
-          element[1][0].trim(),
+          element[1].trim(),
+          null,
           null,
-          element[1][1].trim()
+          comment
         );
       }
-      //else if there is just a type
-      return new Entry(element[0].trim(), element[1].trim(), null, null);
-    }
-    //in case of untyped variable , give default type
-    else {
-      return new Entry(element.trim(), "text", null, null);
-    }
-  }));
+      //in case of untyped variable , give default type
+      else {
+        return new Entry(element.trim(), "text", null, null, comment);
+      }
+    })
+    .filter(element => element.name !== "")); //in case a line started with "#" we would have a blank environmental variable
```

Comment:
It's better to move the filtering before `map`. There is no reason to run operations in `map` for lines that will be ignored.

Let's just filter out lines starting with `#` and then apply `map` to the rest.